scripts/performance/perf_load/perf_processes.py

```diff
@@ -448,6 +453,73 @@ def check_genesis(gen_path):
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(pool.set_protocol_version(2))
+    pool_cfg = json.dumps({"genesis_txn": gen_path})
+    pool_name = "pool_{}".format(random_string(24))
+    loop.run_until_complete(pool.create_pool_ledger_config(pool_name, pool_cfg))
+    pool_handle = loop.run_until_complete(pool.open_pool_ledger(pool_name, None))
+    wallet_cfg = json.dumps({"id": "wallet_{}".format(random_string(24))})
+    wallet_creds = json.dumps({"key": ""})
+    loop.run_until_complete(wallet.create_wallet(wallet_cfg, wallet_creds))
+    wallet_handle = loop.run_until_complete(wallet.open_wallet(wallet_cfg, wallet_creds))
+
+    # read genesis to get aliases and dests
+    with open(dict_args["genesis_path"], 'r') as f:
+        data = f.read()
+        jsons = [json.loads(x) for x in data.split('\n')]
+        aliases = [_json['txn']['data']['data']['alias'] for _json in jsons]
+        dests = [_json['txn']['data']['dest'] for _json in jsons]
+
+    # create stewards
+    names = ['Steward{}'.format(i) for i in range(1, len(jsons) + 1)]
+    seeds = ['{:0>32}'.format(name) for name in names]
+    dids = []
+    for seed in seeds:
+        steward_did, _ = loop.run_until_complete(did.create_and_store_my_did(wallet_handle, json.dumps({'seed': seed})))
+        dids.append(steward_did)
+
+    # put all into list of dicts
+    pool_data = [
+        {'node_alias': node_alias,
+         'steward_did': steward_did,
+         'node_dest': node_dest}
+        for node_alias, steward_did, node_dest in zip(aliases, dids, dests)
+    ]
+
+    async def _demote_promote_periodic():
+        while True:
+            # pick random node from pool to demote/promote it
+            req_data = random.choice(pool_data)
+
+            _data = {
+                'alias': req_data['node_alias'],
+                'services': []
+            }
+            req = await ledger.build_node_request(req_data['steward_did'], req_data['node_dest'], json.dumps(_data))
+            await ledger.sign_and_submit_request(pool_handle, wallet_handle, req_data['steward_did'], req)
+
+            # wait for an interval
+            await asyncio.sleep(interval)
+
+            _data['services'] = ['VALIDATOR']
+            req = await ledger.build_node_request(req_data['steward_did'], req_data['node_dest'], json.dumps(_data))
+            await ledger.sign_and_submit_request(pool_handle, wallet_handle, req_data['steward_did'], req)
+            # restart promoted node
+            host = testinfra.get_host('ssh://persistent_node' + req_data['node_alias'][4:])
+            host.run('sudo systemctl restart indy-node')
+
+    # https://stackoverflow.com/questions/37512182/how-can-i-periodically-execute-a-function-with-asyncio
+    task = loop.create_task(_demote_promote_periodic())
+    loop.call_later(interval * 1000, task.cancel)  # define delay before looping task stopping
+
+    try:
+        loop.run_until_complete(task)
```

Comment:
It waits until the `_demote_promote_periodic` is completed, doesn't it? So, no other requests will be sent in between?